src/systems/InteractionSystem.ts

```diff
@@ -21,9 +23,7 @@ export const InteractionSystem: (
         'position'
       ])[0];
       if (!player) return;
-      const dx = e.position.x - player.position.x;
-      const dy = e.position.y - player.position.y;
-      const distance = Math.sqrt(dx ** 2 + dy ** 2);
+      const distance = dist(e.position, player.position);
       const text = resolveSprite(e.renderable.sprite);
```

Comment:
Once we have a map, we should use a spatially aware index of some sort. The map should know where each entity is in it, for example.